resources/assets/js/components/ApplicantProfile/BasicInfo/ProfileBasicInfo.tsx

```diff
@@ -0,0 +1,495 @@
+  applicantId,
+  currentClassification,
+  currentLevel,
+  citizenshipDeclaration,
+  classifications,
+  email,
+  gcEmployeeStatus,
+  name,
+  previousClassifications,
+  veteranStatus,
+  handleUpdateProfile,
+}) => {
+  const intl = useIntl();
+  const locale = getLocale(intl.locale);
+
+  const defaultValues = {
+    citizenshipDeclaration: citizenshipDeclaration || "",
+    veteranStatus: veteranStatus || "",
+    gcEmployeeStatus: gcEmployeeStatus || "",
+    currentClassification: currentClassification || "",
+    currentLevel: currentLevel || "",
+    previousClassifications,
+  };
+
+  const validationShema = Yup.object().shape({
+    citizenshipDeclaration: Yup.number()
+      .oneOf(
+        Object.values(CitizenshipId),
+        intl.formatMessage(validationMessages.invalidSelection),
+      )
+      .typeError(intl.formatMessage(validationMessages.required))
+      .required(intl.formatMessage(validationMessages.required)),
+    veteranStatus: Yup.number()
+      .oneOf(
+        Object.values(VeteranId),
+        intl.formatMessage(validationMessages.invalidSelection),
+      )
+      .typeError(intl.formatMessage(validationMessages.required))
+      .required(intl.formatMessage(validationMessages.required)),
+    gcEmployeeStatus: Yup.number()
+      .oneOf(
+        Object.values(GCEmployeeStatus),
+        intl.formatMessage(validationMessages.invalidSelection),
+      )
+      .typeError(intl.formatMessage(validationMessages.required))
+      .required(intl.formatMessage(validationMessages.required)),
+    currentClassification: Yup.number().when("gcEmployeeStatus", {
+      is: GCEmployeeStatus.current,
+      then: Yup.number()
+        .oneOf(classifications.map(({ id }) => id))
+        .typeError(intl.formatMessage(validationMessages.required))
+        .required(intl.formatMessage(validationMessages.required)),
+    }),
+    currentLevel: Yup.number().when("gcEmployeeStatus", {
+      is: GCEmployeeStatus.current,
+      then: Yup.number()
+        .min(1, intl.formatMessage(validationMessages.invalidSelection))
+        .max(9, intl.formatMessage(validationMessages.invalidSelection))
+        .typeError(intl.formatMessage(validationMessages.required))
+        .required(intl.formatMessage(validationMessages.required)),
+    }),
+    previousClassifications: Yup.array().of(
+      Yup.object().shape({
+        classification_id: Yup.number()
+          .oneOf(classifications.map(({ id }) => id))
+          .typeError(intl.formatMessage(validationMessages.required))
+          .required(intl.formatMessage(validationMessages.required)),
+        level: Yup.number()
+          .min(1, intl.formatMessage(validationMessages.invalidSelection))
+          .max(9, intl.formatMessage(validationMessages.invalidSelection))
+          .typeError(intl.formatMessage(validationMessages.required))
+          .required(intl.formatMessage(validationMessages.required)),
+      }),
+    ),
+  });
```

Comment:
I know this is how we've done react form validation up until now, and I'm very happy to know that our existing Yup validation objects could be ported directly to Hook Form, but I'm thinking now that this is overkill.
We can control a field being required at the level of HTML, and its probably unnecessary to control for valid values in a Select input. The user would have to edit the values of the fields in the inspector to trigger those errors, at which point we're relying on backend validation anyway.

Reply:
Which means **none** of this Yup validation might be necessary, actually 🤔. (In this case anyway.)